### PR TITLE
Feat/objectarium lzma compression

### PR DESCRIPTION
**Pull request by ErikssonJoakim**
This PR adds the LZMA compression algorithm to the Objectarium smart-contract, a lossless compression with high compression ratio!
- #263 

No tests have been added for the decompression. This could be added at another time with test for snappy decompression as well

### CODE REVIEW

File: contracts/okp4-objectarium/src/compress.rs [hunk 1]
```diff
@@ -77,21 +88,58 @@ fn snappy_decompress(data: &[u8]) -> Result<Vec<u8>, CompressionError> {
...
 }
 
+// lzma_compress returns the LZMA compressed data.
+#[inline]
+fn lzma_compress(data: &[u8]) -> Result<Vec<u8>, CompressionError> {
+    let mut reader = io::Cursor::new(data);
+    let mut writer = Vec::new();
+    {
+        lzma_rs::lzma_compress(&mut reader, &mut writer)?;
+    }
```
Comment (ccamel) [inline contracts/okp4-objectarium/src/compress.rs:98]: No need for parenthesis.

```suggestion
        lzma_rs::lzma_compress(&mut reader, &mut writer)?;
```

File: contracts/okp4-objectarium/src/compress.rs [hunk 2]
```diff
@@ -77,21 +88,58 @@ fn snappy_decompress(data: &[u8]) -> Result<Vec<u8>, CompressionError> {
...
+}
+
+// lzma_decompress returns the LZMA decompressed data.
+#[inline]
+fn lzma_decompress(data: &[u8]) -> Result<Vec<u8>, CompressionError> {
+    let mut reader = io::Cursor::new(data);
+    let mut writer = Vec::new();
+    {
+        lzma_rs::lzma_decompress(&mut reader, &mut writer)?;
+    }
```
Comment (ccamel) [inline contracts/okp4-objectarium/src/compress.rs:109]: No need for parenthesis.

```suggestion
        lzma_rs::lzma_decompress(&mut reader, &mut writer)?;
```

**Review by ccamel — CHANGES_REQUESTED** [review]

**Review by amimart — APPROVED** [review]
Looks good! 😉

**Review by ccamel — APPROVED** [review]
Great! Thanks :+1:

**Pull request merged**